v24.3.0-rc.1 release notes (#19115)
cockroachdb/docs: src/current/_includes/releases/v24.3/v24.3.0-rc.1.md

<h3 id="v24-3-0-rc-1-sql-language-changes">SQL language changes</h3>

- Row-level `AFTER` triggers can now be executed in response to mutations on a table. Row-level `AFTER` triggers fire after checks and cascades have completed for the query. [#133320][#133320]
- Cascades can now execute row-level `BEFORE` triggers. By default, attempting to modify or eliminate the cascading `UPDATE` or `DELETE` operation results in a `Triggered Data Change Violation` error. To bypass this error, you can set the `unsafe_allow_triggers_modifying_cascades` query option to `true`. This could result in constraint violations. [#134444][#134444]
- String constants can now be compared with collated strings. [#134086][#134086]

<h3 id="v24-3-0-rc-1-operational-changes">Operational changes</h3>

- The `kvadmission.low_pri_read_elastic_control.enabled` cluster setting has been removed, because all bulk requests are now subject to elastic admission control admission by default. [#134486][#134486]
- The following metrics have been added for Logic Data Replication (LDR):

>     - `logical_replication.catchup_ranges`: the number of source side ranges conducting catchup scans.

Author: ```suggestion
  - `logical_replication.catchup_ranges`: the number of source side ranges conducting catchup scans.
```

>     - `logical_replication.catchup_ranges`: the number of source side ranges conducting catchup scans.

Author: ```suggestion
  - `logical_replication.catchup_ranges`: the number of source side ranges conducting catchup scans.
```

>     - `logical_replication.scanning_ranges`: the number source side ranges conducting initial scans.

Author: ```suggestion
  - `logical_replication.scanning_ranges`: the number source side ranges conducting initial scans.
```

>     -

Author: ```suggestion
```

>     In the DB Console, these metrics may not be accurate if multiple LDR jobs are running. The metrics are accurate when exported from the Prometheus endpoint. [#134674][#134674]

Author: ```suggestion
  - In the DB Console, these metrics may not be accurate if multiple LDR jobs are running. The metrics are accurate when exported from the Prometheus endpoint. [#134674][#134674]
```

- The `kvadmission.low_pri_read_elastic_control.enabled` cluster setting has been removed, because all bulk requests are now subject to elastic admission control admission by default. [#134486][#134486]
- The following metrics have been added for Logic Data Replication (LDR):
    - `logical_replication.catchup_ranges`: the number of source side ranges conducting catchup scans.
    - `logical_replication.scanning_ranges`: the number source side ranges conducting initial scans.
    -
    In the DB Console, these metrics may not be accurate if multiple LDR jobs are running. The metrics are accurate when exported from the Prometheus endpoint. [#134674][#134674]
- The backup and restore syntax update of `cockroach workload` which was introduced in [#134610][#134610] #has been reverted. [#134645][#134645]

<h3 id="v24-3-0-rc-1-db-console-changes">DB Console changes</h3>

- After finalizing an upgrade to v24.3, an updated version of the **Databases** page will be available. [#134244][#134244]
- Users with the `CONNECT` privilege can now access the **Databases** page. [#134542][#134542]

<h3 id="v24-3-0-rc-1-bug-fixes">Bug fixes</h3>

> - Fixed a bug where an error was logged if an LDAP connection was closed by server and the cluster attempted to use the reset:

Author: ```suggestion
- Fixed a bug where an LDAP connection would be closed by the server and would not be retried by CockroachDB. [#134277]
```

> 

Author: ```suggestion
```

>     ~~~ none

Author: ```suggestion
```

>     LDAP authentication: unable to find LDAP user distinguished name\nerror when searching for user in LDAP server: LDAP search failed: LDAP bind failed: ‹LDAP Result Code 200 \"Network Error\"›: ‹ldap: connection closed›

Author: ```suggestion
```

>     ~~~

Author: ```suggestion
```

> 

Author: ```suggestion
```

>     Closed connections are now handled automatically and the cluster attempts to connect again if the connection was reset. [#134277][#134277]

Author: ```suggestion
- Fixed a bug that prevented LDAP authorization from successfully assigning CockroachDB roles to users when the source group name contained periods or hyphens. [#134944][#134944]
```

[#134086]: https://github.com/cockroachdb/cockroach/pull/134086
[#134089]: https://github.com/cockroachdb/cockroach/pull/134089
[#134097]: https://github.com/cockroachdb/cockroach/pull/134097
[#134244]: https://github.com/cockroachdb/cockroach/pull/134244
[#134267]: https://github.com/cockroachdb/cockroach/pull/134267
[#134277]: https://github.com/cockroachdb/cockroach/pull/134277
[#134367]: https://github.com/cockroachdb/cockroach/pull/134367
[#134427]: https://github.com/cockroachdb/cockroach/pull/134427
[#134444]: https://github.com/cockroachdb/cockroach/pull/134444
[#134448]: https://github.com/cockroachdb/cockroach/pull/134448
[#134482]: https://github.com/cockroachdb/cockroach/pull/134482
[#134486]: https://github.com/cockroachdb/cockroach/pull/134486
[#134510]: https://github.com/cockroachdb/cockroach/pull/134510
[#134536]: https://github.com/cockroachdb/cockroach/pull/134536
[#134542]: https://github.com/cockroachdb/cockroach/pull/134542
[#134645]: https://github.com/cockroachdb/cockroach/pull/134645
[#134674]: https://github.com/cockroachdb/cockroach/pull/134674
[#134675]: https://github.com/cockroachdb/cockroach/pull/134675

> [#134610]: https://github.com/cockroachdb/cockroach/pull/134610

Author: ```suggestion
[#134610]: https://github.com/cockroachdb/cockroach/pull/134610
[#134944]: https://github.com/cockroachdb/cockroach/pull/134944
```
